src/routing/approved.rs

```diff
@@ -993,15 +1015,12 @@ impl Approved {
         }
 
         if join_request.section_key != *self.section.chain().last_key() {
-            let response = BootstrapResponse::Join {
+            let variant = Variant::Rejoin {
```

Comment:
I am still feeling this is bit confusing with the `rejoined node` concept.
I understand here it means `ask the peer to try re-join with new key`.
However, actually the `re-join process` won't help to update the key.
The peer has to go through `re-bootstrap`.

Just wondering, why not use the word of `ReBootstrap`?

Reply:
I see. `Rebootstrap` might work, but it's still not entirely accurate. Let me think about it.

Reply:
How about `JoinRetry` ?